lib/cli-engine/cli-engine.js

```diff
@@ -168,6 +171,7 @@ function calculateStatsPerFile(messages) {
         return stat;
     }, {
         errorCount: 0,
+        fatalErrorCount: 0,
```

Comment:
This adds a new property to [LintResult](https://eslint.org/docs/developer-guide/nodejs-api#-lintresult-type) objects, which our public API returns.

@nzakas is it ok, and should we then document the new property?

Reply:
Yes it’s ok and yes we should document it. Good catch.

Reply:
@A-Katopodis can you add the new property `fatalErrorCount` to the docs for LintResult type? That's [here](https://github.com/eslint/eslint/blob/master/docs/developer-guide/nodejs-api.md#-lintresult-type).

And, since it might be confusing whether or not `errorCount` includes `fatalErrorCount`, I think it would be good to also update the description of the `errorCount` property, to something like:

```
The number of errors. This includes fixable errors and fatal errors.
```

This is only about the ESLint class section, there's no need to update the docs for CLIEngine as it will be removed soon (https://github.com/eslint/eslint/pull/14706).